agr_literature_service/lit_processing/oneoff_scripts/cleanup_resource_data.py

```diff
@@ -0,0 +1,157 @@
+            if x:
+                try:
+                    x.resource_id = correct_resource_id
+                    db_session.add(x)
+                    if row_count % 300 == 0:
+                        db_session.commit()
+                    logger.info("UPDATE EDITOR for editor_id = " + str(editor_id) + " OLD resource_id = " + str(resource_id) + " NEW resource_id = " + str(correct_resource_id))
+                except Exception as e:
+                    logger.info("Error occurred when updating EDITOR for editor_id = " + str(editor_id) + " error = " + str(e))
+
+    db_session.commit()
+
+
+def update_reference_table(db_session, duplicate_resource_id_to_correct_resource_id):
+
+    limit = 5000
+    loop_count = 200
+    row_count = 0
+    for index in range(loop_count):
+        offset = index * limit
+        rows = db_session.execute(f"SELECT reference_id, resource_id FROM reference "
+                                  f"ORDER BY reference_id limit {limit} "
+                                  f"offset {offset}").fetchall()
+        if len(rows) == 0:
+            break
+        for x in rows:
+            reference_id = x[0]
+            resource_id = x[1]
+            if resource_id in duplicate_resource_id_to_correct_resource_id:
+                correct_resource_id = duplicate_resource_id_to_correct_resource_id[resource_id]
+                row_count += 1
+                x = db_session.query(ReferenceModel).filter_by(reference_id=reference_id).one_or_none()
+                if x:
+                    try:
+                        x.resource_id = correct_resource_id
+                        db_session.add(x)
+                        if row_count % 300 == 0:
+                            db_session.commit()
+                        logger.info("UPDATE REFERENCE for reference_id = " + str(reference_id) + " OLD resource_id = " + str(resource_id) + " NEW resource_id = " + str(correct_resource_id))
+                    except Exception as e:
+                        logger.info("Error occurred when updating REFERENCE for reference_id = " + str(reference_id) + " error = " + str(e))
+
+    db_session.commit()
+
+
+def remove_duplicate_ones(db_session, duplicate_resource_id_to_correct_resource_id):
+
+    resource_ids_with_XREF = set()
+    rows = db_session.execute("SELECT resource_id FROM cross_reference "
+                              "WHERE resource_id is not NULL").fetchall()
+    for x in rows:
+        resource_ids_with_XREF.add(x[0])
+
+    row_count = 0
+    for duplicate_resource_id in duplicate_resource_id_to_correct_resource_id:
+        if duplicate_resource_id in resource_ids_with_XREF:
+            logger.info("duplicate_resource_id: " + str(duplicate_resource_id) + " is still in CROSS_REFERENCE table so keep it")
+        else:
+            row_count += 1
+            x = db_session.query(ResourceModel).filter_by(resource_id=duplicate_resource_id).one_or_none()
```

Comment:
@sweng66 : I think @ianlongden thinks we should also delete from resource_version here, based on the duplicate_resource_id

Reply:
I just have a second thought on this.. 

Since this is a such small set comparing with other data in the history tables, I am reluctant to remove them at this moment in its version table - just in case we want to look at some of these data later. 

Of course I can remove these data if you all think we should @azurebrd @ianlongden

Reply:
The version table should have a deleted status at the end I think so we can do the deletion of the version table later on, when you are happy all went okay. In theory at least :-)

Reply:
that sounds good @ianlongden

Reply:
We have A LOT of RANDOM data in the _version tables that we should remove

Reply:
It might be safer if not deleting the version table to dump out the resource_ids of those deleted to a log file as you go along. Just to be extra "safe" that we can clean up the version table later.

Reply:
Actually we should be able to do a left join on the version table later to clean up properly anyway.

Reply:
More precisely from the resource_version table left join to the resource table and delete where no resource found (NULL).

Reply:
I'll defer to you both on what's best.
If 80% of our resource data is bad, removing it from the version table would make sense if it's a lot of data needlessly bloating the database, but since resource objects are not that big, and there's not nearly as many as references, maybe it's just fine to leave it for future cleanup.

Reply:
@azurebrd Yes, just feel more comfortable to keep them for now so we can delete these extra rows when everything is fine.

Reply:
cool, sounds good

Reply:
This is solved.